Handle rusqlite security advistory

Reported in the [cargo-deny job](https://github.com/timberio/vector/runs/620342752). This appears to be a dependency in `trust-dns-resolver`:

```
error[RUSTSEC-2020-0014]: Various memory safety issues

     ┌── /github/workspace/Cargo.lock:327:1 ───
     │
 327 │ rusqlite 0.20.0 registry+https://github.com/rust-lang/crates.io-index
     │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ security vulnerability detected
     │
     = Several memory safety issues have been uncovered in an audit of
       rusqlite.
       
       See https://github.com/rusqlite/rusqlite/releases/tag/0.23.0 for a complete list.
     = URL: https://github.com/rusqlite/rusqlite/releases/tag/0.23.0
     = rusqlite v0.20.0
       └── trust-dns-server v0.17.0
           └── (dev) vector v0.10.0

warning: advisory was not encountered

   ┌── /github/workspace/deny.toml:5:5 ───
   │
 5 │     "RUSTSEC-2019-0031", # spin is unmaintained
   │     ^^^^^^^^^^^^^^^^^^^ no crate matched advisory criteria
   │

2020-04-26 19:18:47 [ERROR] encountered 1 errors
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -192,9 +192,6 @@ tokio01-test = "0.1.1"
 tower-test03 = { package = "tower-test", version = "0.3" }
 tower-test01 = { package = "tower-test", version = "0.1" }
 serde_yaml = "0.8.9"
-trust-dns-server = "0.17.0"
-trust-dns = "0.17.0"
-trust-dns-proto = "0.8.0"
 dirs = "2.0.2"
 tokio-test = "0.2"
 tokio = { version = "0.2", features = ["test-util"] }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ tracing-limit = { path = "lib/tracing-limit" }
 # Tokio / Futures
 futures01 = { package = "futures", version = "0.1.25" }
 futures = { version = "0.3", default-features = false, features = ["compat", "io-compat"] }
-tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing"], default-features = false }
+tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing", "codec"], default-features = false }
 tokio = { version = "0.2.13", features = ["blocking", "fs", "sync", "macros", "test-util", "rt-core", "io-std"] }
 tokio-codec = "0.1.2"
 tokio-openssl = "0.3.0"

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,7 +43,7 @@ tracing-limit = { path = "lib/tracing-limit" }
 # Tokio / Futures
 futures01 = { package = "futures", version = "0.1.25" }
 futures = { version = "0.3", default-features = false, features = ["compat", "io-compat"] }
-tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing", "codec"], default-features = false }
+tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing", "codec", "udp"], default-features = false }
 tokio = { version = "0.2.13", features = ["blocking", "fs", "sync", "macros", "test-util", "rt-core", "io-std"] }
 tokio-codec = "0.1.2"
 tokio-openssl = "0.3.0"

